hbase-server/src/main/java/org/apache/hadoop/hbase/master/normalizer/SimpleRegionNormalizer.java

```diff
@@ -229,6 +231,8 @@ public List<NormalizationPlan> computePlansForTable(final TableDescriptor tableD
       plans.addAll(mergePlans);
     }
 
+    plans = truncateForSize(plans);
```

Comment:
My only other thought is whether it makes sense to do the truncating in the NormalizationWorker, like how it currently handles rate limiting

Reply:
The difference there is that the new setting would apply to HBase clusters using alternate `RegionNormalizer` implementations. I really don't know if that's a pro or a con.

Reply:
Yea, to me the separation of concerns makes sense -- RegionNormalizer generates plans, RegionNormalizerWorker executes those plans based on limits (rate limit and max work limit). The default is unlimited, so even if someone is using their own RegionNormalizer and didn't want limits, they wouldn't get them. But we'd be making the configuration available to all users, which seems like a win with no real downside (given default).

That said, if we decided to keep the shuffling i think that makes sense for SimpleRegionNormalizer to do -- return a shuffled list, rather than have RegionNormalizerWorker shuffle. The shuffling seems ok for the simple normalizer but would probably be unexpected for other implementations.

Reply:
Sure, I've moved the truncation into RegionNormalizerWorker